api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -34,17 +39,33 @@
 	public CohortMembership() {
 	}
 
-	public CohortMembership(Patient patient, Date startDate) {
-		this.patient = patient;
+	public CohortMembership(Integer patientId, Date startDate) {
+		this.patientId = patientId;
 		this.startDate = startDate;
 	}
 
-	public CohortMembership(Patient patient) {
-		this(patient, new Date());
+	public CohortMembership(Integer patientId) {
+		this(patientId, new Date());
 	}
 	
-	public boolean isMemberActive() {
-		return this.getStartDate() != null && !getStartDate().after(new Date()) && this.getEndDate() == null;
+	/**
+	 * @param asOfDate date to compare if membership is active or inactive
+	 * @return boolean true/false if membership is active/inactive
+	 */
+	public boolean isActive(Date asOfDate) {
+		Date date;
+		if (asOfDate == null) {
+			date = new Date();
+		} else {
+			date = asOfDate;
+		}
+		return !this.getVoided()
+				&& (date.equals(this.getStartDate()) || date.after(this.getStartDate()))
```

Comment:
As a Darius suggested, you should using OpenmrsUti.compare(Date, Date) to compare dates